docs/developer-guide/extensions/proxy-extensions.md

```diff
@@ -0,0 +1,190 @@
+# Proxy Extensions
+*Current Status: [Alpha][1] (Since v2.7.0)*
+
+## Overview
+
+With UI extensions it is possible to enhance Argo CD web interface to
+provide valuable data to the user. However the data is restricted to
+the resources that belongs to the Application. With proxy extensions
+it is also possible to add additional functionality that have access
+to data provided by backend services. In this case Argo CD API server
+acts as a reverse-proxy authenticating and authorizing incoming
+requests before forwarding to the backend service.
+
+## Configuration
+
+Proxy extensions are configured in the main Argo CD configmap
+([argocd-cm][2]).
+
+The example below demonstrate all possible configurations available
+for proxy extensions:
+
+```yaml
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: argocd-cm
+  namespace: argocd
+data:
+  extension.config: |
+    extensions:
+    - name: httpbin
+      backend:
+        connectionTimeout: 2s
+        keepAlive: 15s
+        idleConnectionTimeout: 60s
+        maxIdleConnections: 30
+        services:
+        - url: http://httpbin.org
+          cluster: https://some-cluster
```

Comment:
I can't recall... can this be a pattern instead of a hard-coded cluster name?

Reply:
No.. this needs to match exactly what is configured in the application.destination.